### PR TITLE
Fixes and small updates:

### PR DESCRIPTION
**Pull request by MXSH-Dev**
- created an example for adding token validation for user related API calls (will add this to all other restricted API calls later)
- added admin login endpoint place holder
- added admin login UI
- fixed wrong redirecting URL from search bar
- added token expiration, user id, is admin to user login payload and redux state
- added token expiration check and log user out when token is expired
- fixed user not logout properly

#### Next Steps:
1. Add endpoint to search instructor only
2. Discuss course detail API params format `ece220` to `ece/220` 
3. Fix url param for course subject with space in it on frontend after discussion, ie `mec e250`
4. Finish admin login endpoint
5. Finish admin login on frontend
6. Add register admin functionality
7. Add token verification to restricted API calls
8. Add redirects to restricted pages if user not logged in

### CODE REVIEW

File: client/src/components/AsyncSearchBar.tsx [hunk 1]
```diff
@@ -43,15 +43,14 @@ export const AsyncSearchBar: React.FC = () => {
...
     if (option.type == "instructor") {
       routeChange(
-        `instructor/${option.first_name.toLowerCase()}_${option.last_name.toLowerCase()}`,
+        `/instructor/${option.first_name.toLowerCase()}_${option.last_name.toLowerCase()}`,
         history
       );
     } else if (option.type == "course") {
       routeChange(
-        `course/${option.subject_code.toLowerCase()}${option.number}`,
+        `/course/${option.subject_code.toLowerCase()}${option.number}`,
```
Comment (raysarinas) [inline client/src/components/AsyncSearchBar.tsx:53]: should change this to `/course/${option.subject_code.toLowerCase()}/${option.number}` now that Ahmed's changes are merged in

**Review by raysarinas — APPROVED** [review]
LGTM, should probably just change the one thing I made a comment on

File: client/src/components/AsyncSearchBar.tsx (hunk 1 above)
Comment (MXSH-Dev) [reply client/src/components/AsyncSearchBar.tsx:53]: yeah, will do this later. I have it on my list in next steps.

Comment (MXSH-Dev) [reply client/src/components/AsyncSearchBar.tsx:53]: it depends if frontend want to follow the same format I guess...

**Pull request merged**